canonical-websites/snapcraft.io
# Move different flex grid styles into one pattern

Fixes #597 

Consolidates flex-box grid pattern use on snapcraft home page (for logos, linked OS logos, tools cards, featured snaps) into one pattern that can be possibly adapted and proposed to Vanilla.

Once this gets reviewed and merged we should propose it to Vanilla.

Done:
- created new `p-flex-grid` pattern for flex-box based grid
- `p-flex-grid--centred` can be used if grid items are meant to be centered instead of spread in whole width
- added `p-flex-grid__item` and `p-flex-grid__item--small` for 2 sizes 
- moved existing custom components that use flex-box grid to new `p-flex-grid` and items patterns


### QA

- ./run or http://snapcraft.io-pr-1011.run.demo.haus/
- go to snapcraft home page
- make sure all grid components look good in all screen sizes (for component list see https://github.com/canonicalltd/snap-squad/issues/597#issuecomment-412020662)
- go to snapcraft build page
- make sure OS logos look good on all screen sizes
